src/display-api/display-api.service.ts

```diff
@@ -126,4 +133,155 @@ export class DisplayApiService {
       );
     }
   }
+
+  /**
+   * Fetches all screens.
+   */
+  async fetchScreens(): Promise<ScreenScreenJsonld[]> {
+    try {
+      const config = await this.getAuthenticatedConfig();
```

Comment:
You have a lot of these invocations, and they seem to always fetch a token from the API. Don't you run the risk fetching multiple tokens during a single sync? Could we somehow only fetch a token every now and then?

Reply:
Yes, I guess we can do that, would require investigation on the lifetime on such a token.